activerecord/lib/active_record/associations/association.rb

```diff
@@ -53,6 +53,7 @@ def reset
         @target = nil
         @stale_state = nil
         @inversed = false
+        reset_scope
```

Comment:
What about removing `reset_scope` from `reload` since it already calls `reset` ?

Reply:
There are probably also other parts of the codebase which calls both `reset` + `reset_scope` so might be worth checking them (for example the `initializer`)